src/NexusMods.App.UI/Pages/Library/LibraryViewModel.cs

```diff
@@ -0,0 +1,251 @@
+                    current.LinkedLoadoutItems.Add(LibraryLinkedLoadoutItem.FindByLibraryItem(_connection.Db, current.Id.Id));
+
+                    _sourceCache.AddOrUpdate(current);
+                    foreach (var child in current.Children)
+                    {
+                        stack.Push(child);
+                    }
+                }
+            })
+            .OnItemRemoved(node =>
+            {
+                stack.Push(node);
+
+                while (stack.TryPop(out var current))
+                {
+                    _sourceCache.RemoveKey(current.Id);
+                    foreach (var child in current.Children)
+                    {
+                        stack.Push(child);
+                    }
+                }
+            })
+            .DisposeMany()
+            .Bind(out var nodes)
+            .SubscribeWithErrorLogging();
+
+        LibraryLinkedLoadoutItem
+            .ObserveAll(_connection)
+            .Filter(item => item.AsLoadoutItem().LoadoutId == loadoutId)
+            .OnUI()
+            .OnItemAdded(item =>
+            {
+                var lookup = _sourceCache.Lookup(item.LibraryItemId.Value);
+                if (!lookup.HasValue) return;
+
+                lookup.Value.LinkedLoadoutItems.Add(item);
+            })
+            .OnItemRemoved(item =>
+            {
+                var lookup = _sourceCache.Lookup(item.LibraryItemId.Value);
+                if (!lookup.HasValue) return;
+
+                lookup.Value.LinkedLoadoutItems.Remove(item);
+            })
+            .SubscribeWithErrorLogging();
+
+        _sourceCache
+            .Connect()
+            .MergeMany(static node => node.AddToLoadoutCommand)
+            .ToObservable()
+            .Select(_connection, static (node, connection) => node.GetLibraryItemToInstall(connection))
+            .Where(libraryItem => libraryItem.IsValid())
+            .SubscribeAwait(async (libraryItem, cancellationToken) =>
+            {
+                await using var job = _libraryService.InstallItem(libraryItem, Loadout.Load(_connection.Db, loadoutId));
+                await job.StartAsync(cancellationToken: cancellationToken);
+                await job.WaitToFinishAsync(cancellationToken: cancellationToken);
+            }, awaitOperation: AwaitOperation.Parallel, cancelOnCompleted: true, configureAwait: true);
+
+        Source = CreateSource(nodes, createHierarchicalSource: ViewHierarchical);
+
+        this.WhenActivated(disposables =>
+        {
+            this.WhenAnyValue(vm => vm.ViewHierarchical)
+                .Where(viewHierarchical => viewHierarchical && !Source.IsHierarchical)
+                .Select(viewHierarchical => CreateSource(nodes, createHierarchicalSource: viewHierarchical))
+                .BindToVM(this, vm => vm.Source)
+                .DisposeWith(disposables);
+        });
+    }
+
+    private IObservable<IChangeSet<LibraryNode>> ObserveModPages(ILocatableGame game)
```

Comment:
Could this function be renamed when you get around to it?

It may be a leftover, but it messes with my brain, what we're really observing here seems to be `LibraryFile` instances. These are usually mods, so that's fine, but the wording `Pages` is really throwing me off, given that we're not dealing with UI pages here.